File: app/src/main/java/com/example/user/biblejournal/model/asynctasks/GetVerseByIdAsyncTask.java
package com.example.user.biblejournal.model.asynctasks;

import android.os.AsyncTask;

import com.example.user.biblejournal.model.Repository;
import com.example.user.biblejournal.model.database.verse.VerseDao;
import com.example.user.biblejournal.model.database.verse.VerseEntity;

public class GetVerseByIdAsyncTask extends AsyncTask<Integer, Void, VerseEntity> {
    private Repository.VerseRepositoryListener listener;
    private VerseDao mAsyncTaskDao;

    public GetVerseByIdAsyncTask(VerseDao dao, Repository.VerseRepositoryListener listener) {
        mAsyncTaskDao = dao;
        this.listener = listener;
    }

    @Override
    protected VerseEntity doInBackground(final Integer... params) {
        return mAsyncTaskDao.getVerse(params[0], params[1], params[2]);
    }

    @Override
    protected void onPostExecute(VerseEntity verseEntity) {
        super.onPostExecute(verseEntity);

        listener.onVerseRead(verseEntity);
    }
}

File: app/src/main/java/com/example/user/biblejournal/model/asynctasks/GetVerseByIdAsyncTask.kt
package com.example.user.biblejournal.model.asynctasks

import android.os.AsyncTask

import com.example.user.biblejournal.model.Repository
import com.example.user.biblejournal.model.database.verse.VerseDao
import com.example.user.biblejournal.model.database.verse.VerseEntity

class GetVerseByIdAsyncTask(private val mAsyncTaskDao: VerseDao, private val listener: Repository.VerseRepositoryListener) : AsyncTask<Int, Void, VerseEntity>() {

    override fun doInBackground(vararg params: Int?): VerseEntity {
        return mAsyncTaskDao.getVerse(params[0]!!, params[1]!!, params[2]!!)
    }

    override fun onPostExecute(verseEntity: VerseEntity) {
        super.onPostExecute(verseEntity)

        listener.onVerseRead(verseEntity)
    }
}